master/internal/db/postgres_tasks.go

```diff
@@ -209,74 +170,63 @@ func (db *PgDB) StartAllocationSession(
 	}
 	return token, nil
 }
 
 // DeleteAllocationSession deletes the task session with the given AllocationID.
-func (db *PgDB) DeleteAllocationSession(allocationID model.AllocationID) error {
-	_, err := db.sql.Exec(
-		"DELETE FROM allocation_sessions WHERE allocation_id=$1", allocationID)
+func DeleteAllocationSession(ctx context.Context, allocationID model.AllocationID) error {
+	_, err := Bun().NewDelete().Table("allocation_sessions").Where("allocation_id = ?", allocationID).Exec(ctx)
 	return err
 }
 
 // UpdateAllocationState stores the latest task state and readiness.
-func (db *PgDB) UpdateAllocationState(a model.Allocation) error {
-	_, err := db.sql.Exec(`
-		UPDATE allocations
-		SET state=$2, is_ready=$3
-		WHERE allocation_id=$1
-	`, a.AllocationID, a.State, a.IsReady)
+func UpdateAllocationState(ctx context.Context, a model.Allocation) error {
+	_, err := Bun().NewUpdate().Table("allocations").
+		Set("state = ?, is_ready = ?", a.State, a.IsReady).
+		Where("allocation_id = ?", a.AllocationID).Exec(ctx)
+
 	return err
 }
 
 // UpdateAllocationPorts stores the latest task state and readiness.
-func UpdateAllocationPorts(a model.Allocation) error {
+func UpdateAllocationPorts(ctx context.Context, a model.Allocation) error {
 	_, err := Bun().NewUpdate().Table("allocations").
 		Set("ports = ?", a.Ports).
 		Where("allocation_id = ?", a.AllocationID).
-		Exec(context.TODO())
+		Exec(ctx)
 	return err
 }
 
 // UpdateAllocationStartTime stores the latest start time.
-func (db *PgDB) UpdateAllocationStartTime(a model.Allocation) error {
-	_, err := db.sql.Exec(`
-		UPDATE allocations
-		SET start_time = $2
-		WHERE allocation_id = $1
-	`, a.AllocationID, a.StartTime)
+func UpdateAllocationStartTime(ctx context.Context, a model.Allocation) error {
+	_, err := Bun().NewUpdate().Table("allocations").
+		Set("start_time = ?", a.StartTime).Where("allocation_id = ?", a.AllocationID).Exec(ctx)
 	return err
 }
 
 // UpdateAllocationProxyAddress stores the proxy address.
-func (db *PgDB) UpdateAllocationProxyAddress(a model.Allocation) error {
-	_, err := db.sql.Exec(`
-		UPDATE allocations
-		SET proxy_address = $2
-		WHERE allocation_id = $1
-	`, a.AllocationID, a.ProxyAddress)
+func UpdateAllocationProxyAddress(ctx context.Context, a model.Allocation) error {
+	_, err := Bun().NewUpdate().Table("allocations").Set("proxy_address = ?", a.ProxyAddress).
+		Where("allocation_id = ?", a.AllocationID).Exec(ctx)
 	return err
 }
 
 // CloseOpenAllocations finds all allocations that were open when the master crashed
 // and adds an end time.
-func (db *PgDB) CloseOpenAllocations(exclude []model.AllocationID) error {
-	if _, err := db.sql.Exec(`
-	UPDATE allocations
-	SET start_time = cluster_heartbeat FROM cluster_id
-	WHERE start_time is NULL`); err != nil {
+func CloseOpenAllocations(ctx context.Context, exclude []model.AllocationID) error {
+	if _, err := Bun().NewUpdate().Table("allocations", "cluster_id").
```

Comment:
again, maybe i just have to accept this, but the update from syntax for bun reads so weird to me. wdyt?

Reply:
It's a quirk, but I think if we're prioritizing using Bun for the queries, we can use this feature too. 
I'll also leave this comment open, and if more people (@NicholasBlaskey @eecsliu ) also agree that it's hard to understand I'll revert the change.
I don't want to make that call since I'm biased to Bun.

Reply:
I the SQL query looks more legible to me, can we use NewRaw()? 

to be fair a more accurate translation would have been
```
NewUpdate("allocations").
Set("start_time = cluster_heartbeat FROM cluster_id").
Where("start_time IS NULL")
```

but that doesn't help the case where we do actually want to do a ```UPDATE FROM```

Reply:
Okay I'll change the multi-table update/inserts back into NewRaw!